[build] getting error when running make check locally

**Describe the bug**
When building as follow locally(linux):
```
./bootstrap
REFERENCE_DEVICE=1 COVERAGE=1 CFLAGS=-m32 CXXFLAGS=-m32 LDFLAGS=-m32 make -f examples/Makefile-posix check
```
I got the such error:

> /tests/unit/test_lowpan.hpp:63:38: error: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type class ot::TestIphcVector’; use assignment instead [-Werror=class-memaccess]
>    63 |         memset(this, 0, sizeof(*this));
> 

It seems that memset a class object is not allowed.

--- a/tests/unit/test_lowpan.hpp
+++ b/tests/unit/test_lowpan.hpp
@@ -60,7 +60,7 @@ public:
      */
     TestIphcVector(const char *aTestName)
     {
-        memset(this, 0, sizeof(*this));
+        memset(reinterpret_cast<void *>(this), 0, sizeof(TestIphcVector));
         mTestName              = aTestName;
         mSrcContext.mContextId = kContextUnused;
         mDstContext.mContextId = kContextUnused;

